### ISSUE TITLE
Enable rules to parse exif metadata and import as tags

### ISSUE THREAD

**Issue by ghostsquad** [hydrusnetwork/hydrus#1323]
Enable rules to parse exif metadata and import as tags. Possible ideas for how to enable such rules to be described could be inspired by renamer: https://renamer.en.softonic.com/

Or allow some simple scripting to be used such as Javascript or Lua?

Related to: https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/2087

**Comment by Zweibach** [comment]
Since the import system allows you to set what should be imported as what from sidecars (tags, notes, urls) it makes more sense to just apply that to file metadata instead of forcing it all to be tags.

**Comment by ghostsquad** [comment]
> Since the import system allows you to set what should be imported as what from sidecars (tags, notes, urls) it makes more sense to just apply that to file metadata instead of forcing it all to be tags.

My goal, and correct me if I'm wrong about this, is not just to capture the metadata, but sort and organize images based on that information. Essentially bucket images based on text search results. 

Let's say an image prompt is `lara croft on misty mountain`. I need to be some able to find `lara croft`, `mountain`, `misty mountain` and combinations there of.

Hopefully this use case makes sense. As i understand it, that's exactly what tags are meant for.

**Comment by PaperOrb** [comment]
> > Since the import system allows you to set what should be imported as what from sidecars (tags, notes, urls) it makes more sense to just apply that to file metadata instead of forcing it all to be tags.
> 
> My goal, and correct me if I'm wrong about this, is not just to capture the metadata, but sort and organize images based on that information. Essentially bucket images based on text search results.
> 
> Let's say an image prompt is `lara croft on misty mountain`. I need to be some able to find `lara croft`, `mountain`, `misty mountain` and combinations there of.
> 
> Hopefully this use case makes sense. As i understand it, that's exactly what tags are meant for.

Yep, exactly this. It seems like an obviously useful feature to include and would work well for organizing a library of AI generated art that puts the generation parameters into the exif data on the image.
